addon/components/o-s-s/slider.ts

```diff
@@ -53,60 +67,41 @@ export default class SliderComponent extends Component<SliderComponentArgs> {
     this.displayTooltip = false;
   }
 
-  @action
-  onNumberInput(event: InputEvent): void {
-    let value = (event.target as HTMLInputElement).valueAsNumber;
-    if (isNaN(value)) {
-      value = this.sliderOptions.min;
-    }
-    this.updateValue(value);
-  }
-
   @action
   initializeSlider(): void {
     requestAnimationFrame(() => {
-      this.updateBackgroundSize(this.currentRangeValue);
-      this.updateTooltipPosition(this.currentRangeValue);
+      this.updateBackgroundSize();
     });
   }
 
   private getPercentage(value: number): number {
-    return (value - this.sliderOptions.min) / (this.sliderOptions.max - this.sliderOptions.min);
+    return Math.min((value - this.sliderOptions.min) / (this.sliderOptions.max - this.sliderOptions.min), 1);
   }
 
-  private updateValue(value: number): void {
-    this.currentRangeValue = value;
-    this.updateBackgroundSize(value);
-    this.updateTooltipPosition(value);
-    if (this.args.onChange) {
-      this.args.onChange(value);
-    }
+  private updateValue(value: number | null): void {
+    this.args.onChange?.(value);
   }
 
-  private updateBackgroundSize(value: number): void {
-    let percentage = this.getPercentage(value) * 100;
-    const customRangeElement = document.querySelector('.oss-slider--range') as HTMLElement;
-    if (customRangeElement) customRangeElement.style.setProperty('--range-percentage', `${percentage}%`);
+  private updateBackgroundSize(): void {
```

Comment:
Initial quick analysis was :
```
2) oss-component's OSS::Slider
Multiple things to be changed:
- The current slider value is a local copy, we will want it to be based directly on the value in parameter (so that the parent can reset it at will)
    - we can move the variable to a getter which returns this.args.value ?? DEFAULT_VALUE
- Current "background" size is calculated manually at init, and then when the user makes a change
    -We should base the background value off the value getter so that it is dynamic
    - Instead of using a querySelector in the controller and then updating the styles manually, we can pass the styles to the template with a getter ( styles={{this.backgroundStyles}} )
    - We can rename "background" to something more precise, "activeWidth" maybe ?
- As we did for the background above, the tooltip positioning needs to be dynamic instead of manually triggered
```

Reply:
@phndiaye wdyt?

Reply:
It looks like setting a value from outside works properly though, so you guys decide.

Still, I feel like the code would be a lot cleaner with the dynamic style solutions.